packages/icon/src/vaadin-icon.js

```diff
@@ -157,48 +150,39 @@ class Icon extends ThemableMixin(ElementMixin(ControllerMixin(PolymerElement)))
     this.addController(this._tooltipController);
   }
 
-  /** @private */
-  __getIconsetName(icon) {
-    if (!icon) {
-      return;
-    }
-
-    const parts = icon.split(':');
-    return parts[0] || DEFAULT_ICONSET;
-  }
-
-  /** @private */
-  __onIconsetRegistered(e) {
-    if (e.detail === this.__getIconsetName(this.icon)) {
-      this.__iconChanged(this.icon);
-    }
-  }
-
   /** @protected */
   connectedCallback() {
     super.connectedCallback();
-    document.addEventListener('vaadin-iconset-registered', this.__onIconsetRegistered);
```

Comment:
Replaced the event with a different logic for storing instances of icons in the set.